File: 1945-sum-of-digits-of-string-after-convert/1945-sum-of-digits-of-string-after-convert.cpp
class Solution {
public:
    int getLucky(string s, int k) {
        int res = 0;
        for (char c : s) {
            res += c < 'j'   ? c - 'a' + 1
                         : c < 't' ? c - 'a' - 8
                                   : c - 'a' - 17;
        }
        while (--k) {
            int sum = 0;
            while (res) {
                sum += (res % 10);
                res /= 10;
            }
            res = sum;
        }
        return res;
    }
};

File: 1945-sum-of-digits-of-string-after-convert/1945-sum-of-digits-of-string-after-convert.go
func getLucky(s string, k int) int {
	res := 0
	for _, c := range s {
		if c < 'j' {
			res += int(c-'a') + 1
		} else if c < 't' {
			res += int(c-'a') - 8
		} else {
			res += int(c-'a') - 17
		}
	}
	for {
		k--
		if k == 0 {
			break
		}
		sum := 0
		for res > 0 {
			sum += res % 10
			res /= 10
		}
		res = sum
	}
	return res
}